tests/normalize/test_max_nesting.py

```diff
@@ -0,0 +1,290 @@
+    all_table_names = pipeline_schema.data_table_names()
+    assert len(all_table_names) == 1
+    assert all_table_names == [
+        "third_resource_with_nested_data",
+    ]
+
+    # Second we do with max_table_nesting=1
+    some_data_source = some_data_v2()
+    some_data_source.max_table_nesting = 1
+
+    pipeline.drop()
+    pipeline.run(some_data_source, write_disposition="append")
+    pipeline_schema = pipeline.schemas[pipeline.default_schema_name]
+    all_table_names = pipeline_schema.data_table_names()
+    assert len(all_table_names) == 2
+    assert all_table_names == [
+        "third_resource_with_nested_data",
+        "third_resource_with_nested_data__params",
+    ]
+
+    # Second we do with max_table_nesting=2
+    some_data_source = some_data_v2()
+    some_data_source.max_table_nesting = 3
+
+    pipeline.drop()
+    pipeline.run(some_data_source, write_disposition="append")
+    pipeline_schema = pipeline.schemas[pipeline.default_schema_name]
+    all_table_names = pipeline_schema.data_table_names()
+
+    # 5 because payload is a dictionary not a collection of dictionaries
+    assert len(all_table_names) == 5
+    assert all_table_names == [
+        "third_resource_with_nested_data",
+        "third_resource_with_nested_data__payload__hints",
+        "third_resource_with_nested_data__payload__hints__f_float",
+        "third_resource_with_nested_data__payload__hints__f_float__comments",
+        "third_resource_with_nested_data__params",
+    ]
+
+    # Check scenario #3
+    pipeline.drop()
+    some_data_source = some_data()
+    some_data_source.max_table_nesting = 0
+    pipeline.run(some_data_source, write_disposition="append")
+    pipeline_schema = pipeline.schemas[pipeline.default_schema_name]
+    all_table_names = pipeline_schema.data_table_names()
+    assert len(all_table_names) == 5
+    assert sorted(all_table_names) == [
+        "rasa_bot_events_with_nesting_lvl_one",
+        "rasa_bot_events_with_nesting_lvl_two",
+        "rasa_bot_events_with_nesting_lvl_two__metadata__known_recipients",
+        "rasa_bot_events_with_nesting_lvl_two__metadata__vendor_list",
+        "third_resource_with_nested_data",
+    ]
+
+    # Check scenario #4
+    # Set max_table_nesting via the setter and check the tables
+    pipeline.drop()
+    rasa_bot_events_resource = rasa_bot_events_with_nesting_lvl_one()
+    pipeline.run(
+        rasa_bot_events_resource,
+        dataset_name="bot_events",
+        write_disposition="append",
+    )
+    pipeline_schema = pipeline.schemas[pipeline.default_schema_name]
+    all_table_names = pipeline_schema.data_table_names()
+    count_all_tables_first_run = len(all_table_names)
+    tables = pipeline_schema.data_table_names()
+    assert count_all_tables_first_run == 1
+    assert tables == ["rasa_bot_events_with_nesting_lvl_one"]
+
+    # now adjust the max_table_nesting for resource and check
+    pipeline.drop()
+    rasa_bot_events_resource.max_table_nesting = 2
```

Comment:
@rudolfix here I am manually setting/overriding the `max_table_nesting` for the resource.